### PR TITLE
Removed manual padding in resnet v2

### PR DESCRIPTION
**Pull request by zaccharieramzi**
# What does this PR do?

This fixes an underlying issue that was pointed out in #805 (more precisely in [this comment](https://github.com/keras-team/keras-cv/pull/805#issuecomment-1255321847)).

More specifically, in the current resnet v2 implementation, manual padding is used in the convolution with kernel size 3 (i.e. with the class `ZeroPadding2D`). I replaced it with `padding="SAME"` in the `Conv2D` operator, which according to @LukeWood should be faster.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Not exactly but as said this was mentioned in #805 
- [ ] Did you write any new necessary tests? No
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed.

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
Apparently the failed CI test `keras-cv-accelerator-testing (keras-team-test)` is due to a lack of authorization.
I am not sure I can act on it

**Comment by LukeWood** [comment]
Does weight loading still work with this?  Can we make sure the test accuracies on imagenet are identical?

**Comment by tanzhenyu** [comment]
This is fixed during the deeplab development. Closing it for now. Feel free to re-open if you have more questions

**Comment by zaccharieramzi** [comment]
Hi @tanzhenyu ,

What is the deeplab development?

**Comment by tanzhenyu** [comment]
> Hi @tanzhenyu ,
> 
> What is the deeplab development?

https://github.com/keras-team/keras-cv/pull/975/files#diff-cabcb01a9e568b74a40873421782493a0dd75971e73890b5e6c4403050e7c3e0L150
